php_memcached.c

```diff
@@ -576,7 +576,7 @@ static void php_memc_get_impl(INTERNAL_FUNCTION_PARAMETERS, zend_bool by_key)
 	MEMC_METHOD_FETCH_OBJECT;
 	i_obj->rescode = MEMCACHED_SUCCESS;
 
-	if (key_len == 0 || strchr(key, ' ')) {
+	if (key_len == 0 || key_len > 250 || strchr(key, ' ') || strchr(key, '\n')) {
```

Comment:
I would expect it to be:
```
if (key_len == 0 || key_len > 250 || strchr(key, ' ') || strchr(key, '\n' || strchr(key, ' ')) {
```
Therefore, I would suggest to write it something like
if ("key has a character in which is preset in array x" ) {
   do the 'this is not allowed bit.
}

array x would be a constantmadjiggy, define once in the class like
x = {
" ",
"\n"
// and all other characters which are not allowed in the memcachkey.
}